CollAction/Frontend/app/global/NewsletterSignup.tsx

```diff
@@ -0,0 +1,141 @@
+import * as React from "react";
+import renderComponentIf from "./renderComponentIf";
+import { library } from "@fortawesome/fontawesome-svg-core";
+import { faEnvelope } from "@fortawesome/free-solid-svg-icons";
+import * as classnames from "classnames";
+
+library.add(faEnvelope);
+
+enum States { VOID, ERROR, READY, LOADING }
+
+interface NewsLetterSignupProps {
+  mailchimpUser: string;
+  mailChimpListId: string;
+}
+
+interface NewsLetterSignupState {
+  subscriptionState: States;
+  email?: string;
+  signupStatusMessage?: string;
+}
+
+class NewsletterSignup extends React.Component<NewsLetterSignupProps, NewsLetterSignupState> {
+
```

Comment:
Inconsistent newline

Reply:
I'll fix it

Reply:
Done

Reply:
Not pushed yet?

Reply:
I don't see the problem. New line before and after class definition like in other components..

Reply:
Here's a typical react example: https://reactjs.org/docs/components-and-props.html

This is just making sure the formatting is consistent with typical typescript/react code.

Reply:
Would be good to add a newline between the interface definition and the class as well.

Reply:
Ok I updated it.